### PR TITLE
Update to detray version v0.74.2

### PR DESCRIPTION
**Pull request by niermann999**
Update to detray v0.74.2
Main changes:
- Remove distinction between free vector and free track parameters and refactor track param class interface a bit
- Move the transform index onto the barcode
- fixed window navigation (will remove the navigation buffer, as the navigator can now deal with a static fixed size memory)

### CODE REVIEW

**Comment by niermann999** [comment]
I will need to remove an assertion from detray still (the seed generator smears the time, so that it can indeed be negative), but otherwise this should be ready

File: device/common/include/traccc/finding/device/impl/propagate_to_next_surface.ipp [hunk 1]
```diff
@@ -40,18 +38,13 @@ TRACCC_DEVICE inline void propagate_to_next_surface(
...
-
     // Count the number of tracks per seed
     vecmem::device_atomic_ref<unsigned int> num_tracks_per_seed(
         n_tracks_per_seed.at(orig_param_id));
 
     const unsigned int s_pos = num_tracks_per_seed.fetch_add(1);
 
-    if (s_pos >= cfg.max_num_branches_per_seed ||
-        globalIndex >= nav_candidates.size()) {
+    if (s_pos >= cfg.max_num_branches_per_seed || globalIndex >= n_in_params) {
```
Comment (stephenswat) [inline device/common/include/traccc/finding/device/impl/propagate_to_next_surface.ipp:47]: I don't believe this is correct; need to look into this.

**Review by stephenswat — CHANGES_REQUESTED** [review]
I have a concern about one of the changes. Did you check the physics performance of these changes at all? Otherwise I can have a look into it.

File: device/common/include/traccc/finding/device/impl/propagate_to_next_surface.ipp (hunk 1 above)
Comment (niermann999) [reply device/common/include/traccc/finding/device/impl/propagate_to_next_surface.ipp]: Removed

**Review by stephenswat — APPROVED** [review]
After looking into it with Joana I am happy to approve (as long as the CI is also happy).

**Pull request merged**